### PR TITLE
Fix order display frame again

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Made a mistake when i tried to fix this before in that i incremented it even when it shouldn't be incremented! I've added an exception so that the game at least crashes instead of hangs if it gets into trouble.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Should also fix #2267

**Comment by duncanspumpkin** [comment]
I think #2205 is probably this issue as well.

**Comment by duncanspumpkin** [comment]
For reference https://github.com/OpenLoco/OpenLoco/pull/2127 was the PR that almost fixed this. Issue was in this commit that i added :( https://github.com/OpenLoco/OpenLoco/pull/2127/commits/6cc0c2a0b4eaec95bcc706270ad833c2ad08bc78

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 ------------------------------------------------------------------------
 - Fix: [#2379] Text cursor is not rendering in the right position.
 - Fix: [#2380] 'Number of smoothing passes' label is displayed incorrectly.
+- Fix: [#2387] Crash when viewing vehicle orders.
```
Comment (AaronVanGeffen) [inline CHANGELOG.md:5]: This doesn't always happen, though. Can we make it more specific?

Comment (duncanspumpkin) [reply CHANGELOG.md:5]: I could copy the text from 2127

Comment (AaronVanGeffen) [reply CHANGELOG.md:5]: For reference, #2127 added:
> Crash when viewing the orders of a vehicle and ai deletes a train.

That sounds more specific alright. Can I suggest we generalise it to:
```suggestion
- Fix: [#2387] Potential crash when a vehicle is deleted while viewing vehicle orders.
```

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**